modules/heat_conduction/src/kernels/HeatConductionTimeDerivative.C

```diff
@@ -17,7 +17,7 @@ InputParameters validParams<HeatConductionTimeDerivative>()
   // over current volume by setting the use_displaced_mesh flag.
   params.set<bool>("use_displaced_mesh") = true;
 
-  params.addParam<MaterialPropertyName>("specific_heat", "specific_heat",
+  params.addParam<MaterialPropertyName>("specific_heat_name", "specific_heat",
```

Comment:
We need to pick a convention and deprecate the ones that go against it. I still like using _name because it is then clear that you are wanting a property name. In the case where you put a number, the _name indicates to me that I could use a material property. Without the _name I may think that I can't use a property and the parameter is a constant.